ccpp/data/GFS_typedefs.F90

```diff
@@ -2211,6 +2296,25 @@ subroutine sfcprop_create (Sfcprop, IM, Model)
     Sfcprop%soiltype_frac = clear_val
     Sfcprop%lakefrac  = clear_val
     Sfcprop%lakedepth = clear_val
+
+    Sfcprop%use_lake_model = zero
+    if(Model%lkm > 0) then
```

Comment:
@SamuelTrahanNOAA Is it better if we let model stopped under these two cases: (1) use_lake_model=o while lkm is not (2) lkm=o while use_lake_model is not

Reply:
In other words use_lake_model, lkm, and iopt_lake should not be totally independent

Reply:
The `use_lake_model(:)` is an array, not a single flag. It replaces `use_flake`. The value of `use_lake_model(i)` is `lkm` where there are lakes, and `0` elsewhere.

Reply:
Got it. If there is no lake fraction/depth data available, and lkm is set to 1 or 2, will the model be stopped with the error message?

Reply:
No. That would require extra logic at the workflow level. The workflow would have to change LKM based on whether lake data is available. It is better for the model to be able to run without lake data.

Reply:
Can we add a logic in the model level when reading the oro file? Otherwise it will be misleading. Sometimes the model may break up running the lake option without the lake data.

Reply:
Running LKM=1 or 2 without lake data isn't a problem anymore. What broke it was race conditions in threads. That has been fixed.

I'm not sure what you're asking me to do. I don't want the model to abort if lake data is missing because it would break workflows.

Reply:
Okay. I understand now. Thanks.